src/main/java/teammates/ui/template/InstructorFeedbackResultsGroupByQuestionPanel.java

```diff
@@ -0,0 +1,80 @@
+package teammates.ui.template;
+
+import java.util.List;
+
+import teammates.common.util.FieldValidator;
+
+
+/**
+ * Data model for the giver panel in InstructorFeedbackResults for Giver > Question > Recipient,
+ * and for the recipient panel in Recipient > Question > Giver
+ * 
+ *
+ */
+public class InstructorFeedbackResultsGroupByQuestionPanel extends InstructorResultsParticipantPanel {
+    
+    List<InstructorResultsQuestionTable> questionTables;
+    
+    boolean isEmailValid;
+    
+    public InstructorFeedbackResultsGroupByQuestionPanel(List<InstructorResultsQuestionTable> questionTables,
+                                    String profilePictureLink, 
+                                    boolean isGiver, String participantIdentifier, String participantName,
+                                    InstructorResultsModerationButton moderationButton, 
+                                    boolean isModerationButtonDisplayed) {
+        setParticipantIdentifier(participantIdentifier);
+        setName(participantName);
+        setGiver(isGiver);
+        
+        boolean isEmailValid = new FieldValidator()
+                                       .getInvalidityInfo(FieldValidator.FieldType.EMAIL, participantIdentifier).isEmpty();
+        setEmailValid(isEmailValid);
+        this.profilePictureLink = profilePictureLink;
+        
+        this.questionTables = questionTables;
+        
+        setModerationButton(moderationButton);
+        setModerationButtonDisplayed(isModerationButtonDisplayed);
+        
+        setHasResponses(true);
+        
+    }
+    
+    public static InstructorFeedbackResultsGroupByQuestionPanel buildInstructorFeedbackResultsGroupByQuestionPanelWithoutModerationButton(
+                                    List<InstructorResultsQuestionTable> questionTables,
+                                    String profilePictureLink, 
+                                    boolean isGroupedByGiver, String participantIdentifier, String participantName) {
+        return new InstructorFeedbackResultsGroupByQuestionPanel(questionTables, profilePictureLink, isGroupedByGiver, 
+                                                                participantIdentifier, participantName, 
+                                                                null, false);
+    }
+    
+    public static InstructorFeedbackResultsGroupByQuestionPanel buildInstructorFeedbackResultsGroupByQuestionPanelWithModerationButton(
```

Comment:
Should these two methods be implemented as alternate constructors (with different parameters) instead?